runtime/jitserver_launcher/jitserver.c

```diff
@@ -0,0 +1,359 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <dlfcn.h>
+#include <inttypes.h>
+#include <linux/limits.h>
+#include "j9cfg.h"
+#include "j9arch.h"
+#include "jitserver_api.h"
+#include "jitserver_error.h"
+
+#undef JITSERVER_LAUNCHER_DEBUG
+
+#define DIR_SEPARATOR_CHAR '/'
+
+#define PATH_BUFFER_LEN 512
+#define SELF_EXE "/proc/self/exe"
+
+typedef int32_t (*createJITServer)(JITServer **jitServer, void *vm_args);
+
+static void truncateOneLevel(char *path);
+static int32_t append(char **buffer, uint32_t bufLen, char *str);
+static int32_t isDir(const char *path);
+static int32_t isRegularFile(const char *path);
+static int32_t getJvmPath(char *pathBuffer, int32_t pathBufLen);
+static char * getJvmLibPath(void);
+
+static const char *jvmOptions[] = {	"-Xms8m",
```

Comment:
Hard coding options without a way to override them seems like a mistake. I'd add support for -J to pass options through.

Reply:
Currently the jitserver launcher accepts the same arguments that JVM does. I just add the set of `jvmOptions` to the user specified arguments and pass it to JVM. I think, to allow user to override `jvmOptions`, I can just add the user arguments at the end and JVM would pick the the one which comes last.

Reply:
Sounds good.